src/pkcs15init/pkcs15-openpgp.c

```diff
@@ -419,29 +419,42 @@ static int openpgp_store_data(struct sc_pkcs15_card *p15card, struct sc_profile
 	case SC_PKCS15_TYPE_CERT:
 		cinfo = (sc_pkcs15_cert_info_t *) obj->data;
 		cid = &(cinfo->id);
+		unsigned int tag = 0x7F21;
 
 		if (cid->len != 1) {
 			sc_log(card->ctx, "ID=%s is not valid.", sc_dump_hex(cid->value, cid->len));
 			LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_ARGUMENTS);
 		}
 
 		/* OpenPGP card v.2 contains only 1 certificate */
-		if (cid->value[0] != 3) {
+		if (cid->value[0] != 3 && p15card->card->type < SC_CARD_TYPE_OPENPGP_V3) {
 			sc_log(card->ctx,
 			       "This version does not support certificate ID = %d (only ID=3 is supported).",
 			       cid->value[0]);
 			LOG_FUNC_RETURN(card->ctx, SC_ERROR_NOT_SUPPORTED);
 		}
+
+
 		/* Just update the certificate DO */
+		u8 param = (u8) (2 - (cid->value[0] - 1));
```

Comment:
Please explain this calculation in code. If you do not know the specification this looks like magic. Maybe we use constants here instead?

The idea is bascially:
```
AUT: cid-value[0] = 3; param=0
DEC: cid-value[0] = 2; param=1
SIG: cid-value[0] = 1; param=2
```

while I agree with this calculation, the output is 255 for the AUT key-cert-pair in specific cases, and I am not sure why (see next comment). I am not sure yet if it is happening here, we should check param though.

Reply:
done by explaining the calculation (mapping in a comment)